### PR TITLE
Feature: show watched flag for seasons

### PR DESCRIPTION
**Pull request by wutschel**
## Description

This PR adds the possibility to show the watched flag for seasons in the so-called episodesView.

Screenshot: https://ibb.co/QHPYSnb

## Summary for release notes

Feature: show watched flag for seasons

### CODE REVIEW

File: XBMC Remote/DetailViewController.m [hunk 1]
```diff
@@ -3040,6 +3066,16 @@ - (void)layoutSectionView:(UIView*)albumDetailView thumbView:(UIImageView*)thumb
...
     
+    // Add watched overlay icon to lower right corner of thumb, if enabled
+    if (isWatched) {
+        UIImageView *watchedIcon = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"OverlayWatched"]];
+        watchedIcon.frame = CGRectMake(CGRectGetMaxX(thumbImageView.frame) - FLAG_SIZE / 2 - TINY_PADDING,
+                                       CGRectGetMaxY(thumbImageView.frame) - FLAG_SIZE - TINY_PADDING,
+                                       CGRectGetWidth(watchedIcon.frame),
+                                       CGRectGetHeight(watchedIcon.frame));
+        [albumDetailView addSubview:watchedIcon];
+    }
```
Comment (kambala-decapitator) [inline XBMC Remote/DetailViewController.m:3077]: I suspect that this would break in reused headers

Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: Add a fixup to always add the UIImageView and then hide/unhide. Is this what you had in mind?

Comment (kambala-decapitator) [reply XBMC Remote/DetailViewController.m]: Yes, correct.

But I also see that other subviews are always created, although normally you'd create them once and then receive them from a reused cell / header / footer. Maybe their container is anyways re-created? That's not really good, but out of scope of this PR.

Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: True, I also saw this when reading your comment. I am not sure, if the headers are even reused. I can look into this at a later point in time.

Comment (kambala-decapitator) [reply XBMC Remote/DetailViewController.m]: I saw it's called from viewForHeader, so it can be reused

**Review by kambala-decapitator — APPROVED** [review]

File: XBMC Remote/DetailViewController.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/DetailViewController.m]: So we can make this more efficient by using dequeueing, but as it is not implement now we have no problem. Correct?

**Comment by wutschel** [comment]
Squashed and rebased.

File: XBMC Remote/DetailViewController.m (hunk 1 above)
Comment (kambala-decapitator) [reply XBMC Remote/DetailViewController.m]: yes, seems so

**Pull request merged**